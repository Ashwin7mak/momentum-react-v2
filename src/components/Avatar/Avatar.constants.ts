import { PresenceType } from './Avatar.types';

const CLASS_PREFIX = 'md-avatar';

const MAX_INITIALS_SPACE = 1;
const MAX_INITIALS_PERSON = 2;

const COLORS = {
  default: 'default',
  gold: 'gold',
  orange: 'orange',
  lime: 'lime',
  mint: 'mint',
  cyan: 'cyan',
  cobalt: 'cobalt',
  slate: 'slate',
  violet: 'violet',
  purple: 'purple',
  pink: 'pink',
};

const SIZES = {
  24: 24,
  32: 32,
  48: 48,
  72: 72,
  88: 88,
  124: 124,
};

const AVATAR_ICON_SIZE_MAPPING: Record<number, number> = {
  24: 8,
  32: 12,
  48: 14,
  72: 16,
  88: 20,
  124: 28,
};

const TYPES = {
  person: 'person',
  space: 'space',
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  outerWrapper: `${CLASS_PREFIX}-outer-wrapper`,
  iconWrapper: `${CLASS_PREFIX}-icon-wrapper`,
  imageHidden: `${CLASS_PREFIX}-image-hidden`,
};

const DEFAULTS = {
  PRESENCE: PresenceType.Default,
  SIZE: SIZES[24],
  COLOR: COLORS.default,
  TYPE: TYPES.person,
};

export {
  DEFAULTS,
  SIZES,
  STYLE,
  CLASS_PREFIX,
  TYPES,
  MAX_INITIALS_SPACE,
  MAX_INITIALS_PERSON,
  AVATAR_ICON_SIZE_MAPPING,
  COLORS,
};
