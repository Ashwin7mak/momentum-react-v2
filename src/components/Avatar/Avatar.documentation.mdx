Avatar component that can represend either a person or a space.

It supports:

- image
- icon
- initials
- presence icons

Simple usage:

```js
<Avatar title="Chris Adams" />
```
