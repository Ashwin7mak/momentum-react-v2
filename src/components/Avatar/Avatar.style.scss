.md-avatar-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 100vh;

  span {
    color: var(--avatar-color-avatar-text);
  }

  &[data-size="24"] {
    span {
      font-size: 0.75rem;
    }

    width: 1.5em;
    height: 1.5em;
  }

  img {
    border-radius: 100vh;
    height: 100%;
    object-fit: cover;
  }

  // Background Color
  &[data-color=default] {
    background-color: var(--avatar-color-avatar-background-default);
  }

  &[data-color=gold] {
    background-color: var(--avatar-color-avatar-background-gold);
  }

  &[data-color=orange] {
    background-color: var(--avatar-color-avatar-background-orange);
  }

  &[data-color=lime] {
    background-color: var(--avatar-color-avatar-background-lime);
  }

  &[data-color=mint] {
    background-color: var(--avatar-color-avatar-background-mint);
  }

  &[data-color=cyan] {
    background-color: var(--avatar-color-avatar-background-cyan);
  }

  &[data-color=cobalt] {
    background-color: var(--avatar-color-avatar-background-cobalt);
  }

  &[data-color=slate] {
    background-color: var(--avatar-color-avatar-background-slate);
  }

  &[data-color=violet] {
    background-color: var(--avatar-color-avatar-background-violet);
  }

  &[data-color=purple] {
    background-color: var(--avatar-color-avatar-background-purple);
  }

  &[data-color=pink] {
    background-color: var(--avatar-color-avatar-background-pink);
  }

  // Sizes
  &[data-size="24"] {
    span {
      font-size: 0.75rem;
    }

    width: 1.5em;
    height: 1.5em;
  }

  &[data-size="32"] {
    span {
      font-size: 0.875rem;
    }

    width: 2em;
    height: 2em;
  }

  &[data-size="48"] {
    span {
      font-size: 0.75rem;
    }

    width: 3em;
    height: 3em;
  }

  &[data-size="72"] {
    span {
      font-size: 0.75rem;
    }

    width: 4.5em;
    height: 4.5em;
  }

  &[data-size="88"] {
    span {
      font-size: 0.75rem;
    }

    width: 5.5em;
    height: 5.5em;
  }

  &[data-size="124"] {
    span {
      font-size: 0.75rem;
    }

    width: 7.75em;
    height: 7.75em;
  }
}



