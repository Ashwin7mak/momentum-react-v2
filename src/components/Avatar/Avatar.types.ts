export enum PresenceType {
  Default = 'default',
  Active = 'active',
  Meet = 'meet',
  Schedule = 'schedule',
  // Do not Disturb
  DND = 'DND',
  Presenting = 'presenting',
  QuietHours = 'quietHours',
  Away = 'away',
  // Out of Office
  OOO = 'OOO',
}

export type AvatarColor =
  | 'default'
  | 'gold'
  | 'orange'
  | 'lime'
  | 'mint'
  | 'cyan'
  | 'cobalt'
  | 'slate'
  | 'violet'
  | 'purple'
  | 'pink';

export type AvatarSize = 24 | 32 | 48 | 72 | 88 | 124;

export interface Props {
  /**
   * className prop description
   */
  className?: string;

  /**
   * Size of the avatar
   * @default 32
   */
  size?: AvatarSize;

  /**
   * Determines the type of presence the user has. Eg: meet or presenting
   */
  presence?: PresenceType;

  /**
   * URL of image to display
   */
  src?: string;
  /**
   * Alternative text when image cannot be rendered
   */
  alt?: string;
  /**
   * Initials to be displayed on the avatar
   */
  initials?: string;
  /**
   * Name of person/space this avatar is for
   */
  title?: string;
  /**
   * Background color for avatar
   */
  color?: AvatarColor;
  /**
   * Determines if the avatar is for a space or person
   */
  type?: 'person' | 'space';
  /**
   * The icon name to display instead of initials or image
   */
  icon?: string;
}
