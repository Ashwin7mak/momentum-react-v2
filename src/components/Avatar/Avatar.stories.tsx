import React from 'react';
import { Story } from '@storybook/react';

import Avatar, { AvatarProps } from './';

import ThemeProvider from '../ThemeProvider';

export default {
  title: 'Momentum UI/Avatar',
  component: Avatar,
  argTypes: {
    propName: {
      description: 'Description with what the prop does.',
      options: ['Value 1', 'Value 2'],
      control: { type: 'select' }, // change as needed
    },
  },
};

const Template: Story<AvatarProps> = (args) => (
  <ThemeProvider>
    <Avatar {...args} />
  </ThemeProvider>
);

const Story1 = Template.bind({});

Story1.args = {
  src: 'https://images.unsplash.com/photo-1583195764036-6dc248ac07d9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=2855&q=80',
};

const Story2 = Template.bind({});

Story2.args = {
  initials: 'AS',
  color: 'gold',
  size: 24,
};

export { Story1, Story2 };
