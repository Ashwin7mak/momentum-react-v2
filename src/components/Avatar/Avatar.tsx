import React, { useRef, useState, useEffect } from 'react';
import classnames from 'classnames';

import './Avatar.style.scss';
import { PresenceType, Props } from './Avatar.types';
import {
  STYLE,
  DEFAULTS,
  TYPES,
  MAX_INITIALS_PERSON,
  MAX_INITIALS_SPACE,
  AVATAR_PRESENCE_ICON_SIZE_MAPPING,
  AVATAR_ICON_SIZE_MAPPING,
} from './Avatar.constants';
import Icon from '../Icon';
import { getInitials } from './Avatar.utils';

const Avatar: React.FC<Props> = (props: Props) => {
  const {
    className,
    src,
    alt,
    title,
    initials,
    size,
    color,
    presence,
    type = DEFAULTS.TYPE,
    icon,
  } = props;

  const [imageLoaded, setImageLoaded] = useState(false);
  const [imageLoadFailed, setImageLoadFailed] = useState(false);
  const imageRef = useRef<HTMLImageElement>();

  useEffect(() => {
    if (imageRef && imageRef.current && imageRef.current.complete) {
      handleOnLoad();
    }
  }, []);

  useEffect(() => {
    setImageLoadFailed(false);
    setImageLoaded(false);
  }, [src]);

  const renderPresence = (presence: PresenceType) => {
    if (!presence) return;
    if (presence === PresenceType.Default) return;
    let icon: string;
    let fill: string;

    switch (presence) {
      case PresenceType.Active:
        icon = 'unread';
        fill = 'var(--avatar-presence-icon-active)';
        break;
      case PresenceType.Meet:
        icon = 'camera-presence';
        fill = 'var(--avatar-presence-icon-meeting)';
        break;
      case PresenceType.Schedule:
        icon = 'meetings-presence';
        fill = 'var(--avatar-presence-icon-schedule)';
        break;
      case PresenceType.DND:
        icon = 'dnd-presence';
        fill = 'var(--avatar-presence-icon-dnd)';
        break;
      case PresenceType.Presenting:
        icon = 'share-screen';
        fill = 'var(--avatar-presence-icon-presenting)';
        break;
      case PresenceType.QuietHours:
        icon = 'quiet-hours-presence';
        fill = 'var(--avatar-presence-icon-quiet-hours)';
        break;
      case PresenceType.Away:
        icon = 'recents-presence';
        fill = 'var(--avatar-presence-icon-away)';
        break;
      case PresenceType.OOO:
        icon = 'pto-presence';
        fill = 'var(--avatar-presence-icon-ooo)';
        break;
      default:
        break;
    }

    return (
      <div className={STYLE.presenceIconWrapper}>
        <Icon
          name={icon}
          weight="filled"
          fillColor={fill}
          scale={AVATAR_PRESENCE_ICON_SIZE_MAPPING[size]}
        />
      </div>
    );
  };

  const renderInitials = () => {
    if (src) {
      if (!imageLoadFailed && imageLoaded) {
        return null;
      }
    }

    if (icon) {
      return null;
    }

    // Error handling for initials length
    if (initials) {
      type === TYPES.person &&
        initials.length > MAX_INITIALS_PERSON &&
        console.warn(
          `Avatar with type person should not have more than ${MAX_INITIALS_PERSON} initials.`
        );

      type === TYPES.space &&
        initials.length > MAX_INITIALS_SPACE &&
        console.warn(
          `Avatar with type space should not have more than ${MAX_INITIALS_SPACE} initials.`
        );
    }

    return (
      <span>
        {initials
          ? initials
          : title
          ? getInitials(title, type)
          : 'Please provide a title/initials.'}
      </span>
    );
  };

  const renderIcon = () => {
    if (icon) {
      return (
        <Icon
          className={STYLE.iconWrapper}
          name={icon}
          scale={AVATAR_ICON_SIZE_MAPPING[size].scale}
          weight={AVATAR_ICON_SIZE_MAPPING[size].weight}
        />
      );
    }
  };

  const renderImage = () => {
    if (src && !imageLoadFailed) {
      return (
        <img
          src={src}
          alt={alt}
          onLoad={handleOnLoad}
          onError={handleOnError}
          ref={imageRef}
          className={!imageLoaded ? STYLE.imageHidden : ''}
        />
      );
    }
    return null;
  };

  const handleOnLoad = () => {
    setImageLoaded(true);
  };

  const handleOnError = () => {
    setImageLoadFailed(true);
  };

  return (
    <div className="md-avatar-outer-wrapper">
      <div
        className={classnames(STYLE.wrapper, className)}
        data-size={size || DEFAULTS.SIZE}
        data-color={color || DEFAULTS.COLOR}
      >
        {/* Renders by default with initials or title. Doesn't render if src or icon provided */}
        {renderInitials()}
        {/* Renders if src is provided. */}
        {renderImage()}
        {/* Renders if icon is provided. */}
        {renderIcon()}
        {/* Renders if presence is provided. */}
        {renderPresence(presence)}
      </div>
    </div>
  );
};

/**
 * Avatar component that can contain an image or initials
 */

export default Avatar;
