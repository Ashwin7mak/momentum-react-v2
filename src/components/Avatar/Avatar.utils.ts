import { TYPES } from './Avatar.constants';

/**
 * Utility function to convert names into initials
 *
 * @param title The name of the person/space this avatar is for
 * @returns string with initials
 */
export const getInitials = (title: string, type: string): string => {
  if (!title.replace(/\s/g, '').length) return '';
  const letters = [];
  const words = title.trim().split(/ +/);

  letters.push(String.fromCodePoint(words[0].codePointAt(0)));

  if (type !== TYPES.space && words.length > 1) {
    letters.push(String.fromCodePoint(words[words.length - 1].codePointAt(0)));
  }
  return letters.join('');
};
