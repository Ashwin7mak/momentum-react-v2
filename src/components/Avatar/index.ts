import { default as Avatar } from './Avatar';
import { Props } from './Avatar.types';

export type AvatarProps = Props;

export default Avatar;
