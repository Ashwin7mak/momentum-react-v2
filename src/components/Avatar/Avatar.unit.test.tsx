import Avatar from '.';
import { mount } from 'enzyme';
import React from 'react';

describe('Avatar', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      /* ...snapshot test... */
      const wrapper = mount(<Avatar />);
      expect(wrapper).toBeTruthy();
    });
  });

  describe('attributes', () => {
    it('should have X value', () => {
      /* ...attribute tests... */
      const wrapper = mount(<Avatar />);
      expect(wrapper).toBeTruthy();
    });
  });
});
