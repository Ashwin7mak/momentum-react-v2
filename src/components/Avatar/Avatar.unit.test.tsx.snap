// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshot should match snapshot 1`] = `
<Avatar
  title="Cisco Webex"
>
  <div
    className="md-avatar-outer-wrapper"
  >
    <div
      className="md-avatar-wrapper"
      data-color="default"
      data-size={24}
    >
      <span>
        CW
      </span>
    </div>
  </div>
</Avatar>
`;

exports[`Avatar snapshot should match snapshot with initials 1`] = `
<Avatar
  initials="CW"
>
  <div
    className="md-avatar-outer-wrapper"
  >
    <div
      className="md-avatar-wrapper"
      data-color="default"
      data-size={24}
    >
      <span>
        CW
      </span>
    </div>
  </div>
</Avatar>
`;

exports[`Avatar snapshot should match snapshot with presence 1`] = `
<div>
  <Avatar
    key="0"
    presence="default"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="1"
    presence="active"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-active)"
            name="unread"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-active)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="2"
    presence="meet"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-meeting)"
            name="camera-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-meeting)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="3"
    presence="schedule"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-schedule)"
            name="meetings-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-schedule)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="4"
    presence="DND"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-dnd)"
            name="dnd-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-dnd)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="5"
    presence="presenting"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-presenting)"
            name="share-screen"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-presenting)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="6"
    presence="quietHours"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-quiet-hours)"
            name="quiet-hours-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-quiet-hours)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="7"
    presence="away"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-away)"
            name="recents-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-away)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
  <Avatar
    key="8"
    presence="OOO"
    title="Cisco Webex"
  >
    <div
      className="md-avatar-outer-wrapper"
    >
      <div
        className="md-avatar-wrapper"
        data-color="default"
        data-size={24}
      >
        <span>
          CW
        </span>
        <div
          className="md-avatar-presence-icon-wrapper"
        >
          <Icon
            fillColor="var(--avatar-presence-icon-ooo)"
            name="pto-presence"
            weight="filled"
          >
            <div
              className="md-icon-wrapper"
            >
              <svg
                className=""
                data-autoscale={false}
                data-scale={32}
                height="100%"
                style={
                  Object {
                    "fill": "var(--avatar-presence-icon-ooo)",
                  }
                }
                viewBox="0, 0, 32, 32"
                width="100%"
              />
            </div>
          </Icon>
        </div>
      </div>
    </div>
  </Avatar>
</div>
`;

exports[`Avatar snapshot should match snapshot with size 1`] = `
<Avatar
  size={124}
  title="CW"
>
  <div
    className="md-avatar-outer-wrapper"
  >
    <div
      className="md-avatar-wrapper"
      data-color="default"
      data-size={124}
    >
      <span>
        C
      </span>
    </div>
  </div>
</Avatar>
`;

exports[`Avatar snapshot should match snapshot with src & alt 1`] = `
<Avatar
  alt="alt"
  src="src"
  title="CW"
>
  <div
    className="md-avatar-outer-wrapper"
  >
    <div
      className="md-avatar-wrapper"
      data-color="default"
      data-size={24}
    >
      <span>
        C
      </span>
      <img
        alt="alt"
        className="md-avatar-image-hidden"
        onError={[Function]}
        onLoad={[Function]}
        src="src"
      />
    </div>
  </div>
</Avatar>
`;

exports[`Avatar snapshot should match snapshot with type 1`] = `
<Avatar
  title="Cisco Webex"
  type="space"
>
  <div
    className="md-avatar-outer-wrapper"
  >
    <div
      className="md-avatar-wrapper"
      data-color="default"
      data-size={24}
    >
      <span>
        C
      </span>
    </div>
  </div>
</Avatar>
`;
